dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/GenericData.java

```diff
@@ -28,6 +28,11 @@ public GenericData(int size) {
         pairs = new ArrayList<KeyValuePair>(size);
     }
 
+    public void modifyPair(int index, String key, String value) {
+        KeyValueStringPair kvp = new KeyValueStringPair(key, value);
+        pairs.set(index, kvp);
```

Comment:
I think we should have been using pairs.set(...) in all of our setPair methods all along.  @levin-noro this affects what you're working on too.  Should we set all of the setPair methods to use pairs.set instead of pairs.add?  This would eliminate the crazy problem we have with needing to add elements in the right order.  That would also remove the need for this modifyPair method.